proof/x86/correctness/add_mod_256.saw

```diff
@@ -0,0 +1,209 @@
+  crucible_points_to ret_ptr (crucible_term new_ret);
+};
+let cneg_mod_256_alias_ret_a_spec = do {
+  (a, ret_ptr) <- ptr_to_fresh "ret" vec256_type;
+  flag <- crucible_fresh_var "flag" limb_type;
+  p_ptr <- ptr_to_modulus256;
+  lt_modulus256 a;
+  crucible_execute_func [ret_ptr, ret_ptr, crucible_term flag, p_ptr];
+  let new_ret = {{ if (vec256_abs a != 0) && (flag != 0) then vec256_rep (vec256_abs modulus256 - vec256_abs a) else a : Vec256 }};
+  crucible_points_to ret_ptr (crucible_term new_ret);
+};
+
+let {{
+  sub_mod_256 a b = if diff @ 0 then diff + vec256_abs modulus256 else diff
+    where diff = vec256_abs a - vec256_abs b
+}};
+
+// sub_mod_256
+let sub_mod_256_spec = do {
+  ret_ptr <- crucible_alloc vec256_type;
+  (a, a_ptr) <- ptr_to_fresh_readonly "a" vec256_type;
+  (b, b_ptr) <- ptr_to_fresh_readonly "b" vec256_type;
+  p_ptr <- ptr_to_modulus256;
+  lt_modulus256 a;
+  lt_modulus256 b;
+  crucible_execute_func [ret_ptr, a_ptr, b_ptr, p_ptr];
+  let new_ret = {{ vec256_rep (sub_mod_256 a b) : Vec256}};
+  crucible_points_to ret_ptr (crucible_term new_ret);
+};
+let sub_mod_256_alias_ret_a_spec = do {
+  (a, ret_ptr) <- ptr_to_fresh "ret" vec256_type;
+  (b, b_ptr) <- ptr_to_fresh_readonly "b" vec256_type;
+  p_ptr <- ptr_to_modulus256;
+  lt_modulus256 a;
+  lt_modulus256 b;
+  crucible_execute_func [ret_ptr, ret_ptr, b_ptr, p_ptr];
+  new_ret <- crucible_fresh_var "new_ret" vec256_type;
+  let new_ret = {{ vec256_rep (sub_mod_256 a b) : Vec256}};
+  crucible_points_to ret_ptr (crucible_term new_ret);
+};
+let sub_mod_256_alias_ret_b_spec = do {
+  (b, ret_ptr) <- ptr_to_fresh "ret" vec256_type;
+  (a, a_ptr) <- ptr_to_fresh_readonly "a" vec256_type;
+  p_ptr <- ptr_to_modulus256;
+  lt_modulus256 a;
+  lt_modulus256 b;
+  crucible_execute_func [ret_ptr, a_ptr, ret_ptr, p_ptr];
+  let new_ret = {{ vec256_rep (sub_mod_256 a b) : Vec256}};
+  crucible_points_to ret_ptr (crucible_term new_ret);
+};
+let sub_mod_256_alias_a_b_spec = do {
+  ret_ptr <- crucible_alloc vec256_type;
+  (ab, a_ptr) <- ptr_to_fresh_readonly "a" vec256_type;
+  p_ptr <- ptr_to_modulus256;
+  lt_modulus256 ab;
+  crucible_execute_func [ret_ptr, a_ptr, a_ptr, p_ptr];
+  let new_ret = {{ vec256_rep 0 : Vec256}};
+  crucible_points_to ret_ptr (crucible_term new_ret);
+};
+let sub_mod_256_alias_ret_a_b_spec = do {
+  (ab, ret_ptr) <- ptr_to_fresh "ret" vec256_type;
+  p_ptr <- ptr_to_modulus256;
+  lt_modulus256 ab;
+  crucible_execute_func [ret_ptr, ret_ptr, ret_ptr, p_ptr];
+  let new_ret = {{ vec256_rep 0 : Vec256}};
+  crucible_points_to ret_ptr (crucible_term new_ret);
+};
+
+///////////////////////////////////////////////////////////////////////////////
+// Proofs
+///////////////////////////////////////////////////////////////////////////////
+
+// add_mod_256
+verify_x86 "add_mod_256" add_mod_256_spec w4;
+// verify_x86 "add_mod_256" add_mod_256_alias_ret_a_spec w4;
```

Comment:
nit: let's remove the commented lines, alternatively, wrap with `do_prove = false; ...; do_prove = true;`